In [28]:
# # Auxiliares ao desenvolvimento, remover uma vez que o código está funcional
%load_ext autoreload
%autoreload 2
%reload_ext autoreload

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [29]:
import os
import pandas as pd
from maquinas.dados import Dados
from maquinas.graficos import Graficos
from maquinas.motor_cc_shunt import MotorCCShunt

dados = Dados.ler_dados(nome_arquivo='../dados/dados.txt', separador=",")
motor = MotorCCShunt(dados=dados)
graficos = Graficos(caminho_figuras='../figs/')


Objeto MotorCCShunt criado...


In [30]:
motor.mostrar_resultados()

========================= Parâmetros =========================
Uan   =                240.0
Ian   =              10.2549
Ra    =                  2.0
Rf    =                120.0
Rreos =                    0
Ke    =                  0.8
Kt    =                  0.8
J     =                 0.02
RPM   =               1800.0
Pn    =               2500.0
n     =                 85.0
T_L   =                    0
========================= Resultados =========================
fem                  = 219.4902
w_rads               = 188.4956
torque               = 13.2629
fluxo_tensao         = 1.4555
fluxo_torque         = 1.6167
const_magnetica      = 0.7278
corrente_campo       = 2.0000
corrente_armadura    = 10.2549
velocidade           = 188.4956
potencia_mecânica    = 2250.8501
potencia_elétrica    = 2941.1760
rendimento           = 76.5289


In [31]:
motor_2 = MotorCCShunt(dados=dados)
motor_2.ia = 10.2549

motor_2.mostrar_resultados()

Objeto MotorCCShunt criado...
========================= Parâmetros =========================
Uan   =                240.0
Ian   =              10.2549
Ra    =                  2.0
Rf    =                120.0
Rreos =                    0
Ke    =                  0.8
Kt    =                  0.8
J     =                 0.02
RPM   =               1800.0
Pn    =               2500.0
n     =                 85.0
T_L   =                    0
========================= Resultados =========================
fem                  = 219.4902
w_rads               = 188.4956
torque               = 13.2629
fluxo_tensao         = 1.4555
fluxo_torque         = 1.6167
const_magnetica      = 0.7278
corrente_campo       = 2.0000
corrente_armadura    = 10.2549
velocidade           = 188.4956
potencia_mecânica    = 2250.8501
potencia_elétrica    = 2941.1760
rendimento           = 76.5289


# Condições de operação

## Condição 1: Operação nominal (240 V, sem reostato, carga nominal);


In [32]:
motor_cond_1 = MotorCCShunt(dados=dados)
motor.mostrar_resultados()

Objeto MotorCCShunt criado...
========================= Parâmetros =========================
Uan   =                240.0
Ian   =              10.2549
Ra    =                  2.0
Rf    =                120.0
Rreos =                    0
Ke    =                  0.8
Kt    =                  0.8
J     =                 0.02
RPM   =               1800.0
Pn    =               2500.0
n     =                 85.0
T_L   =   11.941130397708886
========================= Resultados =========================
fem                  = 219.4902
w_rads               = 188.4956
torque               = 13.2629
fluxo_tensao         = 1.4555
fluxo_torque         = 1.6167
const_magnetica      = 0.7278
corrente_campo       = 2.0000
corrente_armadura    = 10.2549
velocidade           = 188.4956
potencia_mecânica    = 2250.8501
potencia_elétrica    = 2941.1760
rendimento           = 76.5289




## Condição 2: Subtensão (200 V, carga nominal);



In [33]:
motor_cond_2 = MotorCCShunt(dados=dados)
motor_cond_2.ua = 200
motor_cond_2.mostrar_resultados()

Objeto MotorCCShunt criado...
========================= Parâmetros =========================
Uan   =                  200
Ian   =              10.2549
Ra    =                  2.0
Rf    =                120.0
Rreos =                    0
Ke    =                  0.8
Kt    =                  0.8
J     =                 0.02
RPM   =               1800.0
Pn    =               2500.0
n     =                 85.0
T_L   =                    0
========================= Resultados =========================
fem                  = 179.4902
w_rads               = 188.4956
torque               = 13.2629
fluxo_tensao         = 1.1903
fluxo_torque         = 1.6167
const_magnetica      = 0.7142
corrente_campo       = 1.6667
corrente_armadura    = 10.2549
velocidade           = 188.4956
potencia_mecânica    = 1840.6541
potencia_elétrica    = 2384.3133
rendimento           = 77.1985



## Condição 3: Enfraquecimento de campo (reostato = $50 \Omega $);


In [34]:
motor_cond_3 = MotorCCShunt(dados=dados)
motor_cond_3.r_reos = 50
motor_cond_3.mostrar_resultados()

Objeto MotorCCShunt criado...
========================= Parâmetros =========================
Uan   =                240.0
Ian   =              10.2549
Ra    =                  2.0
Rf    =                120.0
Rreos =                   50
Ke    =                  0.8
Kt    =                  0.8
J     =                 0.02
RPM   =               1800.0
Pn    =               2500.0
n     =                 85.0
T_L   =                    0
========================= Resultados =========================
fem                  = 219.4902
w_rads               = 188.4956
torque               = 13.2629
fluxo_tensao         = 1.4555
fluxo_torque         = 1.6167
const_magnetica      = 1.0310
corrente_campo       = 1.4118
corrente_armadura    = 10.2549
velocidade           = 188.4956
potencia_mecânica    = 2250.8501
potencia_elétrica    = 2799.9995
rendimento           = 80.3875



## Condição 4: Baixa carga (torque reduzido);



In [35]:
motor_cond_4 = MotorCCShunt(dados=dados)
motor_cond_4.torque_carga = 8
motor_cond_4.mostrar_resultados()

Objeto MotorCCShunt criado...
========================= Parâmetros =========================
Uan   =                240.0
Ian   =              10.2549
Ra    =                  2.0
Rf    =                120.0
Rreos =                    0
Ke    =                  0.8
Kt    =                  0.8
J     =                 0.02
RPM   =               1800.0
Pn    =               2500.0
n     =                 85.0
T_L   =                    8
========================= Resultados =========================
fem                  = 219.4902
w_rads               = 188.4956
torque               = 13.2629
fluxo_tensao         = 1.4555
fluxo_torque         = 1.6167
const_magnetica      = 0.7278
corrente_campo       = 2.0000
corrente_armadura    = 6.8703
velocidade           = 188.4956
potencia_mecânica    = 1507.9645
potencia_elétrica    = 2941.1760
rendimento           = 51.2708



## Condição 5: Condição crítica (baixa tensão e alto torque).

In [36]:
motor_cond_5 = MotorCCShunt(dados=dados)
motor_cond_5.torque_carga = 15
motor_cond_5.ua = 200
motor_cond_5.mostrar_resultados()

Objeto MotorCCShunt criado...
========================= Parâmetros =========================
Uan   =                  200
Ian   =              10.2549
Ra    =                  2.0
Rf    =                120.0
Rreos =                    0
Ke    =                  0.8
Kt    =                  0.8
J     =                 0.02
RPM   =               1800.0
Pn    =               2500.0
n     =                 85.0
T_L   =                   15
========================= Resultados =========================
fem                  = 179.4902
w_rads               = 188.4956
torque               = 13.2629
fluxo_tensao         = 1.1903
fluxo_torque         = 1.6167
const_magnetica      = 0.7142
corrente_campo       = 1.6667
corrente_armadura    = 15.7526
velocidade           = 188.4956
potencia_mecânica    = 2827.4334
potencia_elétrica    = 2384.3133
rendimento           = 118.5848
